src/Orleans/Runtime/OutsideRuntimeClient.cs

```diff
@@ -634,7 +639,13 @@ private void SendRequestMessage(GrainReference target, Message message, TaskComp
 
             if (!oneWay)
             {
-                var callbackData = new CallbackData(callback, TryResendMessage, context, message, () => UnRegisterCallback(message.Id), config);
+                var callbackData = new CallbackData(
+                    callback,
+                    tryResendMessage,
+                    context,
+                    message,
+                    unregisterCallback,
```

Comment:
Instead of allocating for the `TrySendMessage` method group and the `UnRegisterCallback` closure, those parameters are allocated once per `InsideRuntimeClient`/`OutsideRuntimeClient` and passed along in every call.